core/services/bulletprooftxmanager/eth_broadcaster.go

```diff
@@ -265,40 +256,48 @@ func (eb *ethBroadcaster) handleInProgressEthTx(etx models.EthTx, attempt models
 		return saveFatallyErroredTransaction(eb.store, &etx)
 	}
 
-	etx.BroadcastAt = &broadcastAt
+	etx.BroadcastAt = &initialBroadcastAt
 
 	if sendError.IsNonceTooLowError() || sendError.IsReplacementUnderpriced() {
-		if isVirginTransaction {
-			// If we get this error, and transaction has never been processed
-			// before, it is extremely likely that an external wallet messed with
-			// our nonce and we have no choice but to send it again.
-			//
-			// On the off chance an external wallet did not do this and we
-			// somehow got a network double-send or something, this fails by
-			// sending the transction twice (one will revert) which is a safe
-			// fail case.
-			return eb.handleExternalWalletUsedNonce(&etx, attempt)
-		}
-		// If this is resuming a previous crashed run, it is likely that our
-		// previous transaction was the one who was confirmed, in which case
-		// we hand it off to the eth confirmer to get the receipt.
+		// There are three scenarios that this can happen:
+		//
+		// SCENARIO 1
 		//
-		// It is however possible an external wallet can have messed with the
-		// account during restart.
+		// This is resuming a previous crashed run. In this scenario, it is
+		// likely that our previous transaction was the one who was confirmed,
+		// in which case we hand it off to the eth confirmer to get the
+		// receipt.
+		//
+		// SCENARIO 2
```

Comment:
It seems that the rebroadcast tx CLI command is now explicitly not covered by the BPTXM, or the state it leaves the database in is essentially identical to scenario 2. (Browsing it quickly it looks like that command does not add a tx or attempt record to the DB.) If in a rush someone accidentally set that command's `endingNonce` flag too high, we would no longer even attempt to recover and could leave the node in a bad state.

Is there a way that we can guard against that?

Reply:
Good point.

We could modify rebroadcast transactions to insert its records into the database, but the problem then becomes that those transactions are now taken under the wing of BPTXM and this can have unintended consequences (you manually created pending txes at a fixed gas price but they are not getting bumped etc).

Alternatively we can manually update the nonce in the database but this has the opposite problem (we have a higher nonce that has transactions not under control of BPTXM, which can cause the node to get stuck).

I cannot think of any obvious way around it. Rebroadcast is a manual tool and we have to assume the user knows what they are doing and is accepting consequences when they run it.

The only thing we can do is warn the user and ask for a confirmation step if they set endingNonce higher than the node's current `NextNonce` value. What do you think?

Reply:
That sounds like a reasonable compromise. I've created a separate story for it here: https://www.pivotaltracker.com/story/show/175127494